Total Score: 80.0

####### flip_flop ########

Cases Failed: 0
Total Cases: 7

Warnings: 2
flip_flop.cpp(11): warning C4456: declaration of 'i' hides previous local declaration
flip_flop.cpp(5): warning C4101: 'num': unreferenced local variable

Compilation Errors: 0

Linker Errors: 0

Runtime Results:
Process ran successfully

Grade: 9
################################

####### leibniz ########

Cases Failed: 0
Total Cases: 8

Warnings: 0

Compilation Errors: 0

Linker Errors: 0

Runtime Results:
Process ran successfully

Grade: 10
################################

####### root_finding ########

Cases Failed: 0
Total Cases: 10

Warnings: 1
c:\users\pavol\source\repos\quickcheck\submissions\homework2\1567275\root_finding.cpp(25) : warning C4701: potentially uninitialized local variable 'f_root' used

Compilation Errors: 0

Linker Errors: 0

Runtime Results:
Process ran successfully

Grade: 9
################################

####### worth_every_penny ########
--- reference
+++ student
@@ -1,40 +1,40 @@
 Case 0:
 1 rings were sold
-Bill's program outputs 0.25
-The exact profit is    0.25
+Bill's program outputs	0.25
+The exact profit is	0.25
 Case 1:
 100 rings were sold
-Bill's program outputs 49525.00
-The exact profit is    49525.00
+Bill's program outputs	49525.00
+The exact profit is	49525.00
 Case 2:
 4000 rings were sold
-Bill's program outputs 79981000.00
-The exact profit is    79981000.00
+Bill's program outputs	79981000.00
+The exact profit is	79981000.00
 Case 3:
 200000 rings were sold
-Bill's program outputs 199999050000.00
-The exact profit is    199999050000.00
+Bill's program outputs	199999050000.00
+The exact profit is	199999050000.00
 Case 4:
 5000000 rings were sold
-Bill's program outputs 124999976250000.00
-The exact profit is    124999976250000.00
+Bill's program outputs	124999976250000.00
+The exact profit is	124999976250000.00
 Case 5:
 10000000 rings were sold
-Bill's program outputs 499999952500000.00
-The exact profit is    499999952500000.00
+Bill's program outputs	499999952500000.00
+The exact profit is	499999952500000.00
 Case 6:
 20000000 rings were sold
-Bill's program outputs 1999999905000000.00
-The exact profit is    1999999905000000.00
+Bill's program outputs	1999999905000000.00
+The exact profit is	1999999905000000.00
 Case 7:
 21222250 rings were sold
-Bill's program outputs 2251919374506672.00
-The exact profit is    2251919374506812.50
+Bill's program outputs	2251919374506672.00
+The exact profit is	2251919374506812.50
 Case 8:
 29999999 rings were sold
-Bill's program outputs 4499999555305432.00
-The exact profit is    4499999557500009.75
+Bill's program outputs	4499999555305432.00
+The exact profit is	4499999557500010.00
 Case 9:
 30000000 rings were sold
-Bill's program outputs 4499999855305422.00
-The exact profit is    4499999857500000.00
+Bill's program outputs	4499999855305422.00
+The exact profit is	4499999857500000.00

Cases Failed: 10
Total Cases: 10

Warnings: 1
worth_every_penny.cpp(21): warning C4018: '<': signed/unsigned mismatch

Compilation Errors: 0

Linker Errors: 0

Runtime Results:
Process ran successfully

Comment: you need to do arithmetic with integers for cents too (as in this case it is luck that cents
are 0.25); you can't cast to double (not enough precision); next time pad using spaces (no grade lost for it)

Grade: 5
################################

